### ISSUE TITLE
Switch to GPL + Commons Clause License

### ISSUE THREAD

**Issue by Alfagun74** [Phalcode/gamevault-app#381]
Our Current GameVault License is CC BY-NC-SA 4.0, which is ideal for what we want, but not really optimized for code. We should think about switching to a code license with the Commons Clause restriction that is most similar to our current license.

**Comment by DACRepair** [comment]
I already saw this coming. All I can say is make sure you are careful about changing licences when moving to a paid model with an open source project (see redis recently). I'm all for the free market etc etc, but this just seems like another step towards closed source / monetization. At best I wish you the best of luck, but in reality this is another push to prevent users from the "allowed circumvention" of your paid features you promised, which is just a glorified federation service. Good luck!

**Comment by Alfagun74** [comment]
No, this isn't another attempt to hinder users from "allowed circumvention." We're simply revising our license terms to better suit code. It's become apparent that our current CC BY-NC-SA 4.0 license isn't ideal to license code and has caused some complications and questions.  

We're only considering transitioning to a code license that closely aligns with the existing terms, such as a copyleft-license with a commons clause combination. Legally, nothing should change; it's merely a move to optimize the license for our code while maintaining the same terms. Our plans for "allowed circumvention" or GameVault+ remain unchanged and have nothing to do with this license update.

**Comment by DACRepair** [comment]
While I want to believe the best intentions, you need to understand that a business must benefit it's board members and/or it's owners. Going commercial means that the assumption of monetization, because that increases profit generally. 

1. Compatibility with other licenses: meaning you have software that is licensed in a way that is compatible with other licenses from dependencies. This is straightforward.

2. The CC doesn't really cover software because it doesn't cover derivative. This means that the binary derivative are not really covered. I could easily circumvent your monetized code and redistribute, and there is not a lot you could do about it. This means you can have control of your monetized code usage, taking away from sales. This is the "to the benefit" part I spoke of. 

While it is easy to assume the worst, and I do agree a better OSS license is needed, I can see how this is driven by business decisions. You have decided to monetize some API functions and some  not on FOSS software, which tells me that you don't really support derivation of your software. Plus "we allow you to do this" doesn't really cut it legally since that can be revoked at any time.

All of this stinks of dishonesty.

**Comment by Alfagun74** [comment]
Alright Sherlock @DACRepair, there seems to be some misunderstanding. GameVault has never been nor was it ever intended to be FOSS. It has always been a free commercial product developed by an indie team of two devs. There are no board members involved.

To understand the reasoning behind the decision not to make it open-source, please refer to [this](https://gamevau.lt/blog/2023/07/13/) and [this](https://gamevau.lt/docs/license-contribute#why-is-gamevault-not-open-source).

> The CC doesn't really cover software because it doesn't cover derivative. This means that the binary derivative are not really covered. I could easily circumvent your monetized code and redistribute, and there is not a lot you could do about it. This means you can have control of your monetized code usage, taking away from sales. This is the "to the benefit" part I spoke of.

This statement is incorrect. Its clearly statedin the first section of the license: 

> Adapted Material means material subject to Copyright and Similar Rights that is **derived** from or based upon the Licensed Material and in which the Licensed Material is translated, altered, arranged, **transformed, or otherwise modified** in a manner requiring permission under the Copyright and Similar Rights held by the Licensor. [...]

I understand your concerns about switching licenses, especially given past instances of rugpulls by larger companies. However, this isn't the case here. There's no need to play detective and search for hidden greedy agendas within our indie company with a yearly total revenue of about 200$ ffs.. 

We aren't gating off essential features with GameVault+, and the decision to consider a license switch stems from user feedback on Discord, expressing concerns about the current license's suitability for code and its potential legal ambiguities.

Have a nice day.
